app/src/components/attachments/ImageUpload.tsx

```diff
@@ -37,12 +38,39 @@ const ImageUpload = () => {
 
   const handleDrop = (e: any) => {
     e.preventDefault();
+
     const files = e.dataTransfer.files;
+
+    const processImage = (image: any) => {
+      const img = new Image();
+      img.src = image;
+
+      img.onload = () => {
+        const canvas = document.createElement('canvas');
+        const ctx = canvas.getContext('2d');
+        const width = img.width;
+        const height = img.height;
+        const aspectRatio = width / height;
+
+        const res = 256; // The largest we want the thumbnail to be is 256 x 256
+        const newWidth = Math.sqrt(res * res * aspectRatio);
+        const newHeight = Math.sqrt((res * res) / aspectRatio);
+        canvas.width = newWidth;
+        canvas.height = newHeight;
+
+        ctx?.drawImage(img, 0, 0, newWidth, newHeight);
+
+        const dataUrl = canvas.toDataURL();
+
+        setImage(dataUrl);
+      };
```

Comment:
Just a nit on PR naming. NERT-361 is focused on upload to S3 buckets. I believe NERT-365 is more correct for this PR

Reply:
Good point. Just renamed it.